test/TestUserAcceptance.hs

```diff
@@ -20,41 +20,48 @@
 
 module Main where
 
-import Data.Char (isAscii, isAlphaNum, chr)
+import Data.Char (isAscii, isAlphaNum)
 import qualified Data.Text as T
+import qualified Data.Text.IO as TextIO
+import qualified Data.Text.Encoding as T
+import qualified Data.Text.ICU as ICU
 import System.IO.Temp (createTempDirectory, getCanonicalTemporaryDirectory)
 import Data.Functor (($>))
 import Data.Ini (parseIni, writeIniFileWith, KeySeparator(..), WriteIniSettings(..))
 import Data.Semigroup ((<>))
 import Data.Either (isRight)
 import Control.Concurrent (threadDelay)
+import Control.Concurrent.STM (atomically)
 import Control.Exception (catch, IOException)
-import System.IO (hPutStr, hPutStrLn, stderr)
+import System.IO (hPutStr, hPutStrLn, stderr, stdout)
 import System.Environment (lookupEnv)
 import System.FilePath.Posix ((</>))
 import Control.Monad (void, when)
 import Data.Maybe (isJust)
+import Data.List (intercalate)
 import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as LB
 import Control.Monad.IO.Class (liftIO)
 import Control.Monad.Reader (runReaderT, ask, ReaderT)
 
 import Control.Lens (Lens', view)
-import Data.List (isInfixOf, intercalate)
-import System.Process (callProcess, readProcess)
+import System.Process.Typed
+       (proc, runProcess_, withProcess_,
+        waitExitCodeSTM, setStdout, getStdout, setStderr, useHandleOpen,
+        byteStringOutput, setStdin, closed, ProcessConfig)
 import System.Directory
        (getCurrentDirectory, removeDirectoryRecursive, removeFile, copyFile)
 import Test.Tasty (TestTree, TestName, defaultMain, testGroup, withResource)
 import Test.Tasty.HUnit (testCaseSteps, assertBool)
-import Text.Regex.Posix ((=~))
 
 import Data.MIME (parse, message, mime, MIMEMessage)
 
 {-# ANN module ("HLint: ignore Reduce duplication" :: String) #-}
 
 -- | A condition to check for in the output of the program
 data Condition
-  = Literal String
-  | Regex String
+  = Literal T.Text
+  | Regex T.Text
```

Comment:
Wasn't sure if I should keep it `T.Text` or use `Regex`. I did like the fact, that I can easily just add more text on the `buildAnsiRegex` return value. What do you think?